integration-libs/opf/base/core/facade/opf-global-functions.service.ts

```diff
@@ -204,23 +204,24 @@ export class OpfGlobalFunctionsService implements OpfGlobalFunctionsFacade {
           MerchantCallback
         ] = [submitSuccess, submitPending, submitFailure];
 
-        return this.opfPaymentFacade
-          .submitPayment({
-            additionalData,
-            paymentSessionId,
-            cartId,
-            callbackArray,
-            paymentMethod,
-            returnPath: undefined,
-          })
-          .pipe(
-            finalize(() => {
-              if (overlayedSpinner) {
-                this.stopLoaderSpinner(overlayedSpinner);
-              }
+        return lastValueFrom(
+          this.opfPaymentFacade
+            .submitPayment({
+              additionalData,
+              paymentSessionId,
+              cartId,
+              callbackArray,
+              paymentMethod,
+              returnPath: undefined,
             })
-          )
-          .toPromise();
```

Comment:
Because `toPromise` (https://rxjs.dev/deprecations/to-promise) is deprecated I had to refactor two functions in this file. They need to be verified if works as expected.

Reply:
Code change is good to me.
Tested succefully, 'finalize' gets calle as expecte and spinner gets closed as expected
![image](https://github.com/SAP/spartacus/assets/30024415/d1888b54-14fe-438b-9f0a-55eff1cf5116)